src/main/starlark/builtins_bzl/common/java/java_library.bzl

```diff
@@ -180,3 +205,10 @@ java_library = rule(
     compile_one_filetype = [".java"],
     toolchains = [semantics.JAVA_TOOLCHAIN],
 )
+
+def java_library(**kwargs):
+    _java_library(
+        # Skip over the leading '@'.
+        current_repository = repository_name()[1:],
```

Comment:
`repository_name()[1:]` creates a new string and will probably cause a memory regression, `repository_name()` is probably ok. If possible process it, that is remove `@` in an action.

Reply:
I can definitely change that, but don't we get the regression either way? `repository_name` is backed by [`getNameWithAt`](https://cs.opensource.google/bazel/bazel/+/b750f8c0242d7fcb581d368d8b75e59c51c13a61:src/main/java/com/google/devtools/build/lib/packages/StarlarkNativeModule.java;l=510), so the call will return a new string each time. The returned value isn't retained if we strip off the `@`.